packages/agreement-process/src/services/agreementStateProcessor.ts

```diff
@@ -144,3 +170,181 @@ export const suspendedByProducerFlag = (
   requesterOrgId === agreement.producerId
     ? destinationState === agreementState.suspended
     : agreement.suspendedByProducer;
+
+const allowedStateTransitions = (state: AgreementState): AgreementState[] =>
+  match(state)
+    .with(agreementState.draft, agreementState.pending, () => [
+      agreementState.missingCertifiedAttributes,
+    ])
+    .with(agreementState.missingCertifiedAttributes, () => [
+      agreementState.draft,
+    ])
+    .with(agreementState.active, () => [agreementState.suspended])
+    .with(agreementState.suspended, () => [
+      agreementState.active,
+      agreementState.suspended,
+    ])
+    .with(agreementState.archived, agreementState.rejected, () => [])
+    .exhaustive();
+
+const updatableStates = Object.values(agreementState).filter(
+  (state) => allowedStateTransitions(state).length > 0
+);
+
+async function updateAgreementState(
```

Comment:
is this function actually async?

Reply:
Actually, it's not 👍  Removing async and promise.all